pkg/resources/scim_integration.go

```diff
@@ -74,140 +86,170 @@ func SCIMIntegration() *schema.Resource {
-		return fmt.Errorf("error creating security integration")
+	req.WithSyncPassword(d.Get("sync_password").(bool))
+	if v, ok := d.GetOk("comment"); ok {
+		req.WithComment(v.(string))
+	}
+	if err := client.SecurityIntegrations.CreateScim(ctx, req); err != nil {
+		return diag.FromErr(err)
 	}
 
-	d.SetId(name)
+	d.SetId(helpers.EncodeSnowflakeID(id))
 
-	return ReadSCIMIntegration(d, meta)
+	return ReadContextSCIMIntegration(ctx, d, meta)
 }
 
-// ReadSCIMIntegration implements schema.ReadFunc.
-func ReadSCIMIntegration(d *schema.ResourceData, meta interface{}) error {
+// ReadContextSCIMIntegration implements schema.ReadFunc.
+func ReadContextSCIMIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
-	id := d.Id()
-
-	stmt := snowflake.NewSCIMIntegrationBuilder(id).Show()
-	row := snowflake.QueryRow(db, stmt)
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
 
-	// Some properties can come from the SHOW INTEGRATION call
-
-	s, err := snowflake.ScanScimIntegration(row)
+	integration, err := client.SecurityIntegrations.ShowByID(ctx, id)
 	if err != nil {
-		return fmt.Errorf("could not show security integration")
+		log.Printf("[DEBUG] notification integration (%s) not found", d.Id())
+		d.SetId("")
+		return diag.FromErr(err)
 	}
 
-	// Note: category must be Security or something is broken
-	if c := s.Category.String; c != "SECURITY" {
-		return fmt.Errorf("expected %v to be an Security integration, got %v", id, c)
+	if c := integration.Category; c != "SECURITY" {
+		return diag.FromErr(fmt.Errorf("expected %v to be a SECURITY integration, got %v", id, c))
 	}
 
-	if err := d.Set("scim_client", strings.TrimPrefix(s.IntegrationType.String, "SCIM - ")); err != nil {
-		return err
+	if err := d.Set("name", integration.Name); err != nil {
+		return diag.FromErr(err)
 	}
 
-	if err := d.Set("name", s.Name.String); err != nil {
-		return err
+	if err := d.Set("comment", integration.Comment); err != nil {
+		return diag.FromErr(err)
 	}
 
-	if err := d.Set("created_on", s.CreatedOn.String); err != nil {
-		return err
+	if err := d.Set("created_on", integration.CreatedOn.String()); err != nil {
+		return diag.FromErr(err)
 	}
 
-	// Some properties come from the DESCRIBE INTEGRATION call
-	// We need to grab them in a loop
-	var k, pType string
-	var v, unused interface{}
-	stmt = snowflake.NewSCIMIntegrationBuilder(id).Describe()
-	rows, err := db.Query(stmt)
-	if err != nil {
-		return fmt.Errorf("could not describe security integration")
+	if err := d.Set("enabled", integration.Enabled); err != nil {
+		return diag.FromErr(err)
 	}
-	defer rows.Close()
-	for rows.Next() {
-		if err := rows.Scan(&k, &pType, &v, &unused); err != nil {
-			return fmt.Errorf("unable to parse security integration rows")
+	typeParts := strings.Split(integration.IntegrationType, "-")
```

Comment:
this logic should happen on the SDK side

Reply:
Moved.